File: ac-cli/src/platforms/amp.ts
import { BasePlatform } from './base.js';
import type { CopyOptions } from '../types.js';

export class AmpPlatform extends BasePlatform {
  name = 'amp';
  configFile = 'AGENTS.md';
  skillsDir = '.agents/skills';
  userSkillsDir = '~/.config/agents/skills';
  subagentsDir = '';
  supportsSkills = true;
  supportsSubagents = false;

  async copyAgents(
    source: string,
    target: string,
    options: CopyOptions
  ): Promise<void> {
    await super.copyAgents(source, target, options);
  }

  async copySkill(
    skillName: string,
    sourceDir: string,
    targetDir: string,
    options: CopyOptions
  ): Promise<void> {
    await super.copySkill(skillName, sourceDir, targetDir, options);
  }

  async copySubagent(
    _agentName: string,
    _sourceDir: string,
    _targetDir: string,
    _options: CopyOptions
  ): Promise<void> {
    console.log('Amp does not support subagents. Skipping subagent copy.');
  }

  async validate(target: string) {
    const result = await super.validate(target);

    const skillsDir = this.getSkillPath('', target, false);
    const skills = await (await import('../utils/file.js')).listDirs(skillsDir);

    for (const skill of skills) {
      const skillPath = this.getSkillPath(skill, target, false);
      const skillMdPath = `${skillPath}/SKILL.md`;

      const fileExists = await (await import('../utils/file.js')).fileExists(skillMdPath);
      if (!fileExists) {
        result.warnings.push(`Amp skill ${skill} missing SKILL.md`);
        continue;
      }

      const content = await (await import('../utils/file.js')).readFile(skillMdPath);
      if (!content.startsWith('---')) {
        result.warnings.push(`Amp skill ${skill} missing YAML frontmatter`);
      }

      if (!content.includes('name:')) {
        result.warnings.push(`Amp skill ${skill} missing 'name' in frontmatter`);
      }

      if (!content.includes('description:')) {
        result.warnings.push(`Amp skill ${skill} missing 'description' in frontmatter`);
      }
    }

    return result;
  }
}

File: ac-cli/internal/platform/amp.go
package platform

import (
	"fmt"
	"path/filepath"
	"strings"

	"github.com/sebastiaanwouters/ac-cli/internal/config"
	"github.com/sebastiaanwouters/ac-cli/internal/fileutil"
)

type AmpPlatform struct {
	*BasePlatform
}

func NewAmpPlatform() *AmpPlatform {
	return &AmpPlatform{
		BasePlatform: NewBasePlatform(
			"amp",
			"AGENTS.md",
			".agents/skills",
			"~/.config/agents/skills",
			"",
			true,
			false,
		),
	}
}

func (p *AmpPlatform) CopySubagent(_, _, _ string, _ config.CopyOptions) error {
	fmt.Println("Amp does not support subagents. Skipping subagent copy.")
	return nil
}

func (p *AmpPlatform) Validate(target string) (*config.ValidationResult, error) {
	result, err := p.BasePlatform.Validate(target)
	if err != nil {
		return nil, err
	}

	skillsDir := p.GetSkillPath("", target, false)
	skills, err := fileutil.ListDirs(skillsDir)
	if err != nil {
		return nil, err
	}

	for _, skill := range skills {
		skillPath := p.GetSkillPath(skill, target, false)
		skillMdPath := filepath.Join(skillPath, "SKILL.md")

		if !fileutil.FileExists(skillMdPath) {
			result.Warnings = append(result.Warnings, fmt.Sprintf("Amp skill %s missing SKILL.md", skill))
			continue
		}

		content, err := fileutil.ReadFile(skillMdPath)
		if err != nil {
			return nil, err
		}

		if !strings.HasPrefix(content, "---") {
			result.Warnings = append(result.Warnings, fmt.Sprintf("Amp skill %s missing YAML frontmatter", skill))
		}

		if !strings.Contains(content, "name:") {
			result.Warnings = append(result.Warnings, fmt.Sprintf("Amp skill %s missing 'name' in frontmatter", skill))
		}

		if !strings.Contains(content, "description:") {
			result.Warnings = append(result.Warnings, fmt.Sprintf("Amp skill %s missing 'description' in frontmatter", skill))
		}
	}

	return result, nil
}
